services/uwsgi.ini

```diff
@@ -16,9 +18,26 @@ chmod-socket = 666
 logto = /var/log/uwsgi/%n.log
 log-format = %(addr) - - [%(ltime)] "%(method) %(uri) %(proto)" %(status) %(size) "%(referer)" "%(uagent)" host_hdr=%(host) req_time_elapsed=%(msecs)
 
-processes = 6
+; max number of workers; %k is the number of detected CPU cores
+processes = %(%k + 1)
 threads = 2
+; forcefully kill workers after 300s
 harakiri = 300
 lazy-apps = true
-touch-reload = %(base)/uwsgi.ini
+touch-workers-reload = %(base)/uwsgi.ini
 buffer-size = 131072
+
+; minimum of 2 workers at any time
+; the master process will spawn and shut down workers as needed
+cheaper = 2
+; start off with four workers at start up
+cheaper-initial = 3
+; increase by one worker if needed
+cheaper-step = 1
+; algorithm to use to create a worker; spare is the default
+cheaper-algo = spare
```

Comment:
This is the default algorithm for cheaper but explicitly added it here. The idea is to spawn a new worker when it is needed and to attempt to kill them gracefully.